grails-app/services/org/pih/warehouse/inventory/RecordStockProductInventoryTransactionService.groovy

```diff
@@ -0,0 +1,49 @@
+package org.pih.warehouse.inventory
+
+import org.pih.warehouse.api.AvailableItem
+import org.pih.warehouse.core.Constants
+
+import grails.gorm.transactions.Transactional
+
+@Transactional
+class RecordStockProductInventoryTransactionService extends ProductInventoryTransactionService<RecordInventoryCommand> {
+
+    InventoryService inventoryService
+
+    Transaction createAdjustmentTransaction(
+            RecordInventoryCommand recordInventoryCommand,
+            Date transactionDate
+    ) {
+        Transaction transaction = new Transaction(recordInventoryCommand.properties)
+        transaction.transactionType = TransactionType.get(Constants.ADJUSTMENT_CREDIT_TRANSACTION_TYPE_ID)
+        transaction.inventory = recordInventoryCommand.inventory
+        transaction.comment = recordInventoryCommand.comment
+        transaction.transactionNumber = inventoryService.generateTransactionNumber(transaction)
+        transaction.transactionDate = transactionDate
+        return transaction
+    }
+
+    TransactionEntry createTransactionEntry(
```

Comment:
can you name this createAdjustmentTransactionEntry just for clarity? I originally made the ProductInventoryTransactionService class for the baseline transaction only, so it'd be good to clarify that this method is for adjustments only